python/lsst/daf/butler/core/webdavutils.py

```diff
@@ -50,49 +52,116 @@ def getHttpSession() -> requests.Session:
     """
+
+    retries = Retry(total=3, backoff_factor=1, status_forcelist=[429, 500, 502, 503, 504])
+
     session = requests.Session()
+    session.mount("http://", HTTPAdapter(max_retries=retries))
+    session.mount("https://", HTTPAdapter(max_retries=retries))
+
     log.debug("Creating new HTTP session...")
 
     try:
-        env_auth_method = os.environ['WEBDAV_AUTH_METHOD']
+        env_auth_method = os.environ['LSST_BUTLER_WEBDAV_AUTH']
     except KeyError:
-        raise KeyError("Environment variable WEBDAV_AUTH_METHOD is not set, please use values X509 or TOKEN")
+        raise KeyError("Environment variable LSST_BUTLER_WEBDAV_AUTH is not set, \
+                      please use values X509 or TOKEN")
 
     if env_auth_method == "X509":
+        log.debug("... using x509 authentication.")
         try:
-            proxy_cert = os.environ['WEBDAV_PROXY_CERT']
+            proxy_cert = os.environ['LSST_BUTLER_WEBDAV_PROXY_CERT']
         except KeyError:
-            raise KeyError("Environment variable WEBDAV_PROXY_CERT is not set")
-        log.debug("... using x509 authentication.")
+            raise KeyError("Environment variable LSST_BUTLER_WEBDAV_PROXY_CERT is not set")
         session.cert = (proxy_cert, proxy_cert)
     elif env_auth_method == "TOKEN":
-        try:
-            bearer_token = os.environ['WEBDAV_BEARER_TOKEN']
-        except KeyError:
-            raise KeyError("Environment variable WEBDAV_BEARER_TOKEN is not set")
         log.debug("... using bearer-token authentication.")
-        session.headers = CaseInsensitiveDict({'Authorization': 'Bearer ' + bearer_token})
+        refreshToken(session)
     else:
-        raise ValueError("Environment variable WEBDAV_AUTH_METHOD must be set to X509 or TOKEN")
+        raise ValueError("Environment variable LSST_BUTLER_WEBDAV_AUTH must be set to X509 or TOKEN")
+
+    ca_bundle = None
+    try:
+        ca_bundle = os.environ['LSST_BUTLER_WEBDAV_CA_BUNDLE']
+    except KeyError:
+        log.warning("Environment variable LSST_BUTLER_WEBDAV_CA_BUNDLE is not set: \
+HTTPS requests will fail. If you intend to use HTTPS, please \
+export this variable.")
+
+    session.verify = ca_bundle
+
+    # This header is required for request redirection, in dCache for example
+    if "LSST_BUTLER_WEBDAV_EXPECT100" in os.environ:
+        log.debug("Expect: 100-Continue header enabled.")
+        session.headers.update({'Expect': '100-continue'})
 
     log.debug("Session configured and ready.")
 
     return session
 
 
+def isTokenAuth() -> bool:
+    """Returns the status of bearer-token authentication.
+
+    Returns
+    -------
+    isTokenAuth : `bool`
+        True if LSST_BUTLER_WEBDAV_AUTH is set to TOKEN, False otherwise.
+    """
+    try:
+        env_auth_method = os.environ['LSST_BUTLER_WEBDAV_AUTH']
+    except KeyError:
+        raise KeyError("Environment variable LSST_BUTLER_WEBDAV_AUTH is not set, \
```

Comment:
Again, use string continuation as above in log message. Not a backslash within the string.